embedded/store/preconditions.go

```diff
@@ -108,8 +108,8 @@ func (cs *PreconditionKeyNotModifiedAfterTx) Validate(st *ImmuStore) error {
 	return nil
 }
 
-func (cs *PreconditionKeyNotModifiedAfterTx) Check(idx *indexer) (bool, error) {
-	_, tx, _, err := idx.Get(cs.Key)
+func (cs *PreconditionKeyNotModifiedAfterTx) Check(idx KeyIndex) (bool, error) {
+	valRef, err := idx.Get(cs.Key)
```

Comment:
Deleted entries are considered not modified after given TX - I'm not sure if this is a good approach.

Deletion of an entry could result in valid constraint check that leads to removal of some intermediate changes made to the key.

e.g.:

* Key has value X set at tx 1, two writers read that value
* Writer 1 adds update to the key in tx 2 (using precondition not modified since tx 1)
* Someone deletes the key in tx 3
* Writer 2 updates the key with precondition not modified since tx 1 - this succeeds

Shouldn't writer 2 receive an error that the constraint has failed?

Reply:
aham, probably the constraint should support the spec if deleted ones are ignored or not, it would be consistent with the optional filtering when reading and scanning over keys

Reply:
yup, agree, however I think that the default behavior would be to consider deletion as entry modification. That way you can still do atomic operations on the database that also consider deletion as one of operations.

This also has strange corner cases when we consider data expiration. I think that by default it should not matter if the entry expired or not. The important fact would be if it was set as expirable after given TX.

In case of non-indexable entries I don't think we can support such case.